jax/_src/numpy/lax_numpy.py

```diff
@@ -4624,6 +4638,15 @@ def _is_int_arraylike(x):
           or issubdtype(getattr(x, "dtype", None), np.integer)
           or isinstance(x, (list, tuple)) and _all(_is_int_arraylike(e) for e in x))
 
+def _is_int_scalar_like(x):
```

Comment:
Why not separate the "is scalar" aspect from the "type" aspect? i.e., revert the part of the change about checking types, and only have the part of the change about checking scalar-ness.

I'm wondering if you could just define scalarness something like this:
```
def _is_scalar(x):
  return  np.isscalar(x) or (isinstance(x, ndarray) and np.ndim(x) == 0)
```

Reply:
I replaced these with
```
_is_scalar(x) and issubdtype(_dtype(x), np.integer)
```